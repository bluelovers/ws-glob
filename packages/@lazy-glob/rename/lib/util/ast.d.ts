import { IPartAstInterface, IQuestionMark, IAsterisk } from '../types';
export declare function isAsterisk(g: any | IPartAstInterface): g is IAsterisk;
export declare function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark;
