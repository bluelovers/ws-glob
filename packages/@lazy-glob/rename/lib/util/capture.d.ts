export declare function capture(names: string[], glob: string): import("./globGroupsCollection").globGroupsCollectionInterface[];
