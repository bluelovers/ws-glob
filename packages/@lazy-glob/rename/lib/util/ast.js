"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isQuestionMark = exports.isAsterisk = void 0;
const types_1 = require("../types");
function isAsterisk(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.Asterisk);
}
exports.isAsterisk = isAsterisk;
function isQuestionMark(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.QuestionMark);
}
exports.isQuestionMark = isQuestionMark;
//# sourceMappingURL=ast.js.map