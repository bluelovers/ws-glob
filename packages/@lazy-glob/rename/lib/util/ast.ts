import { IPartAstInterface, IQuestionMark, EnumWildcard, IAsterisk } from '../types';

export function isAsterisk(g: any | IPartAstInterface): g is IAsterisk
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.Asterisk);
}

export function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.QuestionMark);
}
