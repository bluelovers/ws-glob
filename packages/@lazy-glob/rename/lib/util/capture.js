"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * For each string in names array, return a globGroupsCollection (an object implementing globGroupsCollectionInterface) containing the substring matched by each part of the glob pattern.
 * Example of parts of a glob: "?omer.*" => Parts: ["?"] ["omer."] ["*"]
 * @method capture
 * @param names {String[]} List of names
 * @param glob {String} The glob pattern to be applied to names
 * @return {globGroupsCollection[] | null} An array of globGroupsCollection objects. Returns null if invalid parameters or names list is empty.
 */
const globGroupsCollection_1 = require("./globGroupsCollection");
function capture(names, glob) {
    if (!Array.isArray(names) || !names.length) {
        return null;
    }
    if ((typeof glob !== 'string') || !glob.length) {
        return null;
    }
    // For each name received, create an object of type 'globGroupsCollection' containing the match for each part of the glob
    return names.map(function buildGlobGroupsCollection(name) {
        let groupsObj = globGroupsCollection_1.globGroupsCollectionFactory();
        groupsObj.initGroups(glob);
        groupsObj.buildGroups(name);
        return groupsObj;
    });
}
//# sourceMappingURL=capture.js.map