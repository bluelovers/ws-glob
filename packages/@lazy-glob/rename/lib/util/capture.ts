/**
 * For each string in names array, return a globGroupsCollection (an object implementing globGroupsCollectionInterface) containing the substring matched by each part of the glob pattern.
 * Example of parts of a glob: "?omer.*" => Parts: ["?"] ["omer."] ["*"]
 * @method capture
 * @param names {String[]} List of names
 * @param glob {String} The glob pattern to be applied to names
 * @return {globGroupsCollection[] | null} An array of globGroupsCollection objects. Returns null if invalid parameters or names list is empty.
 */
import { globGroupsCollectionFactory } from './globGroupsCollection';

function capture(names: string[], glob: string)
{
	if (!Array.isArray(names) || !names.length)
	{
		return null;
	}

	if ((typeof glob !== 'string') || !glob.length)
	{
		return null;
	}

	// For each name received, create an object of type 'globGroupsCollection' containing the match for each part of the glob
	return names.map(function buildGlobGroupsCollection(name)
	{
		let groupsObj = globGroupsCollectionFactory();
		groupsObj.initGroups(glob);
		groupsObj.buildGroups(name);
		return groupsObj;
	});
}
