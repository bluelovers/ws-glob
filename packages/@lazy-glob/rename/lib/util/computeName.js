"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.computeName = void 0;
const deconstruct_1 = require("./deconstruct");
const capture_1 = require("./capture");
const countWildcards_1 = require("./countWildcards");
const types_1 = require("../types");
/**
 * Builds new names based on provided glob patterns.
 * @param filenames {String[] | string} List of original names, or a single name (string), to compute new names for
 * @param srcGlob {String} Glob pattern used to match the original names
 * @param dstGlob {String} Glob pattern used to contruct new names
 * @return {String[]} List of new names. String is empty if new name could not be computed.
 * @throws {Error} An Error object
 */
function computeName(filenames, srcGlob, dstGlob) {
    if ((typeof srcGlob !== 'string' || !srcGlob) || (typeof dstGlob !== 'string' || !dstGlob)) {
        throw new TypeError('Invalid type for glob pattern! Must be a non-empty string.');
    }
    if (typeof filenames === 'string') {
        filenames = [filenames];
    }
    else {
        if (Array.isArray(filenames)) {
            if (!filenames.length) {
                throw new TypeError('Invalid names list. Array must be non-empty.');
            }
        } // names is not an array, nor a string
        else {
            throw new TypeError('Invalid type for names! Must be a string or an Array of string.');
        }
    }
    // Deconstruct the source glob into literal and wildcard parts
    let srcGlobParts = (0, deconstruct_1.deconstruct)(srcGlob, { collapse: true });
    if (!srcGlobParts.length) {
        throw new Error('Unexpected error while parsing glob.');
    }
    // Deconstruct the destination glob into literal and wildcard parts
    let dstGlobParts = (0, deconstruct_1.deconstruct)(dstGlob, { collapse: false });
    if (!dstGlobParts.length) {
        throw new Error('Unexpected error while parsing glob.');
    }
    let srcWildcardsCount = (0, countWildcards_1.countWildcards)(srcGlobParts);
    let dstWildcardsCount = (0, countWildcards_1.countWildcards)(dstGlobParts);
    // Check if destination glob has more * and/or ? wildcards than source glob does...
    if ((dstWildcardsCount.stars > srcWildcardsCount.stars) || (dstWildcardsCount.questions > srcWildcardsCount.questions)) {
        throw new Error('Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)');
    }
    // extract matches for each wildcard (and literal) parts of the source glob
    let srcCaptureGroupsArray = (0, capture_1.capture)(filenames, srcGlob);
    if (!srcCaptureGroupsArray) {
        throw new Error('Unexpected error while extracting glob matches.');
    }
    let computedNames = filenames.map(function buildNewName(name, nameIdx) {
        // The source glob is required to match the original name in order to proceed building the new name
        if (!srcCaptureGroupsArray[nameIdx].hasMatch()) {
            return '';
        }
        let newName = '';
        let srcAsteriskList = srcCaptureGroupsArray[nameIdx].getAsterisk();
        let srcAsteriskIterator = (srcAsteriskList === null || srcAsteriskList === void 0 ? void 0 : srcAsteriskList.length) ? srcAsteriskList.entries() : null;
        let srcQuestionMarkList = srcCaptureGroupsArray[nameIdx].getQuestionMark();
        let srcQuestionMarkIterator = (srcQuestionMarkList === null || srcQuestionMarkList === void 0 ? void 0 : srcQuestionMarkList.length)
            ? srcQuestionMarkList.entries()
            : null;
        // Cycle through the parts of the destination glob to build the new name
        for (const destPart of dstGlobParts) {
            let srcGroup;
            switch (destPart) {
                case types_1.EnumWildcard.Asterisk:
                    srcGroup = srcAsteriskIterator.next();
                    // value[0] is the key, value[1] is the value, e.g. the actual captureGroup object.
                    // Append the * match to the new name construction
                    newName = newName + srcGroup.value[1].match;
                    break;
                case types_1.EnumWildcard.QuestionMark:
                    srcGroup = srcQuestionMarkIterator.next();
                    // value[0] is the key, value[1] is the value, e.g. the actual captureGroup object.
                    // Appends the ? match to the new name construction
                    newName = newName + srcGroup.value[1].match;
                    break;
                default: // For literal parts, simply append them to new name construction
                    newName = newName + destPart;
            }
        }
        return newName;
    });
    return computedNames;
}
exports.computeName = computeName;
//# sourceMappingURL=computeName.js.map