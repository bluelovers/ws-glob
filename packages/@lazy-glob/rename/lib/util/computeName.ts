import { deconstruct } from './deconstruct';
import { capture } from './capture';
import { countWildcards } from './countWildcards';
import { EnumWildcard, IPartsPattern } from '../types';

/**
 * Builds new names based on provided glob patterns.
 * @param filenames {String[] | string} List of original names, or a single name (string), to compute new names for
 * @param srcGlob {String} Glob pattern used to match the original names
 * @param dstGlob {String} Glob pattern used to contruct new names
 * @return {String[]} List of new names. String is empty if new name could not be computed.
 * @throws {Error} An Error object
 */
export function computeName(filenames: string | string[], srcGlob: string, dstGlob: string)
{
	if ((typeof srcGlob !== 'string' || !srcGlob) || (typeof dstGlob !== 'string' || !dstGlob))
	{
		throw new TypeError('Invalid type for glob pattern! Must be a non-empty string.');
	}
	if (typeof filenames === 'string')
	{
		filenames = [filenames];
	}
	else
	{
		if (Array.isArray(filenames))
		{
			if (!filenames.length)
			{
				throw new TypeError('Invalid names list. Array must be non-empty.');
			}
		} // names is not an array, nor a string
		else
		{
			throw new TypeError('Invalid type for names! Must be a string or an Array of string.');
		}
	}

	// Deconstruct the source glob into literal and wildcard parts
	let srcGlobParts: IPartsPattern[] = deconstruct(srcGlob, { collapse: true });
	if (!srcGlobParts.length)
	{
		throw new Error('Unexpected error while parsing glob.');
	}

	// Deconstruct the destination glob into literal and wildcard parts
	let dstGlobParts: IPartsPattern[] = deconstruct(dstGlob, { collapse: false });
	if (!dstGlobParts.length)
	{
		throw new Error('Unexpected error while parsing glob.');
	}

	let srcWildcardsCount = countWildcards(srcGlobParts);
	let dstWildcardsCount = countWildcards(dstGlobParts);

	// Check if destination glob has more * and/or ? wildcards than source glob does...
	if ((dstWildcardsCount.stars > srcWildcardsCount.stars) || (dstWildcardsCount.questions > srcWildcardsCount.questions))
	{
		throw new Error('Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)');
	}

	// extract matches for each wildcard (and literal) parts of the source glob
	let srcCaptureGroupsArray = capture(filenames, srcGlob);
	if (!srcCaptureGroupsArray)
	{
		throw new Error('Unexpected error while extracting glob matches.');
	}

	let computedNames = filenames.map(function buildNewName(name, nameIdx)
	{
		// The source glob is required to match the original name in order to proceed building the new name
		if (!srcCaptureGroupsArray[nameIdx].hasMatch())
		{
			return '';
		}

		let newName = '';
		let srcAsteriskList = srcCaptureGroupsArray[nameIdx].getAsterisk();
		let srcAsteriskIterator = srcAsteriskList?.length ? srcAsteriskList.entries() : null;
		let srcQuestionMarkList = srcCaptureGroupsArray[nameIdx].getQuestionMark();
		let srcQuestionMarkIterator = srcQuestionMarkList?.length
			? srcQuestionMarkList.entries()
			: null;

		// Cycle through the parts of the destination glob to build the new name
		for (const destPart of dstGlobParts)
		{
			let srcGroup;
			switch (destPart)
			{
				case EnumWildcard.Asterisk:
					srcGroup = srcAsteriskIterator.next();
					// value[0] is the key, value[1] is the value, e.g. the actual captureGroup object.
					// Append the * match to the new name construction
					newName = newName + srcGroup.value[1].match;
					break;
				case EnumWildcard.QuestionMark:
					srcGroup = srcQuestionMarkIterator.next();
					// value[0] is the key, value[1] is the value, e.g. the actual captureGroup object.
					// Appends the ? match to the new name construction
					newName = newName + srcGroup.value[1].match;
					break;
				default: // For literal parts, simply append them to new name construction
					newName = newName + destPart;
			}
		}
		return newName;
	});

	return computedNames;
}
