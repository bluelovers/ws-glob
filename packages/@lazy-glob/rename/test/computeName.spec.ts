import { computeName } from '../lib/util/computeName';

describe('When parameters of invalid type are passed', function ()
{
	describe('renamer.computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should throw a TypeError', function ()
		{
			// @ts-ignore
			expect(() => computeName({}, '*.txt', '*.out')).toThrowErrorMatchingSnapshot();

			// @ts-ignore
			expect(() => computeName(1, '*.txt', '*.out')).toThrowErrorMatchingSnapshot();

			// @ts-ignore
			expect(() => computeName(['bob-smith'], [], '???-*')).toThrowErrorMatchingSnapshot();

			// @ts-ignore
			expect(() => computeName('bob-smith', '*-???', 0)).toThrowErrorMatchingSnapshot();

			// @ts-ignore
			expect(() => computeName(['bob-smith'], '', {})).toThrowErrorMatchingSnapshot();

			expect(() => computeName(['bob-smith'], '???-*', '')).toThrowErrorMatchingSnapshot();

			// @ts-ignore
			expect(() => computeName('bob-smith', 1, '')).toThrowErrorMatchingSnapshot();

			expect(() => computeName([], '*', '*')).toThrowErrorMatchingSnapshot();

		});
	});
});

/*
            Source glob doesn't match
 */
describe('When source glob does not match anything', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return empty string as result', function ()
		{
			let result = computeName(['alx-rose.red', '_1234'], '*.????', '*');
			expect(result[0]).toHaveLength(0);
			expect(result[1]).toHaveLength(0);

			result = computeName('^$', '?', '?');
			expect(result[0]).toHaveLength(0);
			result = computeName('.', '**??', '*?');
			expect(result[0]).toHaveLength(0);
			result = computeName('123.456', '*_*', '*');
			expect(result[0]).toHaveLength(0);
			result = computeName('^abc$', '???b??*', '???');
			expect(result[0]).toHaveLength(0);
			result = computeName('1234', '123', '6789');
			expect(result[0]).toHaveLength(0);
			result = computeName('1234', '12345', '6789');
			expect(result[0]).toHaveLength(0);
			result = computeName('1234', '*2**5', '**6789');
			expect(result[0]).toHaveLength(0);
			result = computeName('1234', '12?45', '?6789');
			expect(result[0]).toHaveLength(0);
		});
	});
});

/*
            # of wildcards in SG < in DG
 */
describe('When number of given wildcard in destination glob is higher than in source glob', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{

		it('should throw an Error', function ()
		{
			/************ # of ? in SG < in DG ************/

			[
				[
					'alx-rose.red', '*', '*?'
				],
				[
					'^$', '??', '???'
				],
				[
					'^bruce.wayne$', '*?*', '??'
				],
				[
					'^bruce.wayne$', '*.wayne?', '??'
				],
				[
					'uvwxyz', 'uvwxyz', '123456?'
				],
			].forEach((argv) => {
				// @ts-ignore
				expect(() => computeName(...argv)).toThrowErrorMatchingSnapshot();
			});





			/************ # of * in SG < in DG ************/

			[
				[
					'.', '*', '*.*'
				],
				[
					'.', '?', '?*'
				],
				[
					// ** is src glob counts as * ; but ** in dst glob remains unchanged
					'abc', '**', '**'
				],
				[
					// ***b* -> *b* in src glob; but *** in dst glob remains unchanged
					'abc', '***b*', '***'
				],
				[
					'abc', '*ab**', '***'
				],
				[
					'uvwxyz', 'uvwxyz', '123456*'
				],
			].forEach((argv) => {
				// @ts-ignore
				expect(() => computeName(...argv)).toThrowErrorMatchingSnapshot();
			});

		});
	});
});

/*
            # wildcard in SG >= in DG
 */
describe('When number of given wildcard in destination glob is equal or fewer than in source glob', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return computed new name', function ()
		{
			/************ # of ? in SG >= in DG ************/
			let result = computeName('alx-rose.red', '*?', '?onald');
			expect(result[0]).toBe('donald');
			result = computeName('.', '*?*', '?');
			expect(result[0]).toBe('.');
			result = computeName('abcdef', '*??*', '?');
			expect(result[0]).toBe('e');
			result = computeName('^$', '?*?', '??');
			expect(result[0]).toBe('^$');
			result = computeName('^abc$', '?abc?', '?123?');
			expect(result[0]).toBe('^123$');
			result = computeName('^abc$', '**??b??*', '???');
			expect(result[0]).toBe('^ac');
			result = computeName('^abc$', '*??b??*', '?*?*?');
			expect(result[0]).toBe('^ac');
			result = computeName('.', '?**', '?');
			expect(result[0]).toBe('.');

			/************ # of * in SG >= in DG ************/
			result = computeName('.', '*?', '?*');
			expect(result[0]).toBe('.');
			result = computeName(['.'], '*?', '*');
			expect(result[0]).toBe('');
			result = computeName('123456', '*?*', '**?');
			expect(result[0]).toBe('123456');
			result = computeName('abc', '*a*b*c*', '***');
			expect(result[0]).toBe('');
			result = computeName(['abcdef', '123bf4'], '**b**f**', '***');
			expect(result[0]).toBe('acde');
			expect(result[1]).toBe('1234');

			/*********** cases with wildcards in path **********/
			result = computeName('test/test-data/123.doc', '*t/?es?-**/*.???*', 'test/*?-*2/????.**');
			expect(result[0]).toBe('test/test-data2/tdoc.123');
		});
	});
});

/*
            When source glob is a literal
 */
describe('When source glob is a literal', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return literally the dst glob, only if it does not contain any wildcard', function ()
		{
			let result = computeName('uvwxyz', 'uvwxyz', '123456');
			expect(result[0]).toBe('123456');

			// but should return empty string if src glob does not match
			result = computeName('uvwxyz', 'uvwxy', '12345');
			expect(result[0]).toHaveLength(0);
		});
	});
});

/*
            When destination glob is a literal
 */
describe('When destination glob is a literal', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return a literally the dst glob, the source glob matches', function ()
		{
			let result = computeName('uvwxyz', '*', '123456');
			expect(result[0]).toBe('123456');
			result = computeName('uvwxyz', 'abc', '123456');
			expect(result[0]).toHaveLength(0);
			result = computeName('uvwxyz', '???', '123456');
			expect(result[0]).toHaveLength(0);
			result = computeName('uvwxyz', '*u', '123456');
			expect(result[0]).toHaveLength(0);
			result = computeName('uvwxyz', '*z', '123456');
			expect(result[0]).toBe('123456');
		});
	});
});

/*
            When names is an array of string
 */
describe('When names is an array of string', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return an array of new names', function ()
		{
			let result = computeName(['abcde', '1234', '!@#$%^&'], '*', 'uvwxyz');
			expect(result.length).toBe(3);
			expect(result[0]).toBe('uvwxyz');
			expect(result[1]).toBe('uvwxyz');
			expect(result[2]).toBe('uvwxyz');
			result = computeName(['abcde', '1234', '!@#$%^&'], '*', '*');
			expect(result.length).toBe(3);
			expect(result[0]).toBe('abcde');
			expect(result[1]).toBe('1234');
			expect(result[2]).toBe('!@#$%^&');
			result = computeName(['abcde', '1234', '!@#$%^&'], '???', '123???');
			expect(result.length).toBe(3);
			expect(result[0]).toHaveLength(0);
			expect(result[1]).toHaveLength(0);
			expect(result[2]).toHaveLength(0);
			result = computeName(['abcde', '1234', '!@#$%^&'], '*????', '*');
			expect(result.length).toBe(3);
			expect(result[0]).toBe('a');
			expect(result[1]).toBe('');
			expect(result[2]).toBe('!@#');
			result = computeName(['abcde', '1234', '!@#$%^&'], '???????', '???????');
			expect(result.length).toBe(3);
			expect(result[0]).toHaveLength(0);
			expect(result[1]).toHaveLength(0);
			expect(result[2]).toBe('!@#$%^&');
			result = computeName(['abcde', '1234', '!@#$%^&'], '!@#$%^&', 'xyz');
			expect(result.length).toBe(3);
			expect(result[0]).toHaveLength(0);
			expect(result[1]).toHaveLength(0);
			expect(result[2]).toBe('xyz');
			result = computeName(['123', '5637', '893', 'abc'], '??3', '?-?');
			expect(result.length).toBe(4);
			expect(result[0]).toBe('1-2');
			expect(result[1]).toHaveLength(0);
			expect(result[2]).toBe('8-9');
			expect(result[3]).toHaveLength(0);
			result = computeName(['123', '5637', '3', 'a3cd'], '??3*', '?-?-*');
			expect(result.length).toBe(4);
			expect(result[0]).toBe('1-2-');
			expect(result[1]).toBe('5-6-7');
			expect(result[2]).toHaveLength(0);
			expect(result[3]).toHaveLength(0);
		});
	});
});

/*
            When "names" is a string
 */
describe('When string is passed in "names" argument', function ()
{
	describe('computeName(names, srcGlob, dstGlob)', function ()
	{
		it('should return a result array with a single element', function ()
		{
			let result = computeName('axl-rose', '*', '*');
			expect(result.length).toBe(1);
			expect(result[0]).toBe('axl-rose');

			result = computeName('', '?', '?');
			expect(result.length).toBe(1);
			expect(result[0]).toHaveLength(0);
		});
	});
});
