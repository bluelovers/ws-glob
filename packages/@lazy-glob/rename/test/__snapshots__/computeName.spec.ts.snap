// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 1`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 2`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 3`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 4`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 5`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 6`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 7`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 8`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 9`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 10`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When number of given wildcard in destination glob is higher than in source glob computeName(names, srcGlob, dstGlob) should throw an Error 11`] = `"Invalid glob pattern. Destination glob contains incorrect number or type of wildcard. (** is treated as * in source glob.)"`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 1`] = `"Invalid type for names! Must be a string or an Array of string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 2`] = `"Invalid type for names! Must be a string or an Array of string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 3`] = `"Invalid type for glob pattern! Must be a non-empty string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 4`] = `"Invalid type for glob pattern! Must be a non-empty string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 5`] = `"Invalid type for glob pattern! Must be a non-empty string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 6`] = `"Invalid type for glob pattern! Must be a non-empty string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 7`] = `"Invalid type for glob pattern! Must be a non-empty string."`;

exports[`When parameters of invalid type are passed renamer.computeName(names, srcGlob, dstGlob) should throw a TypeError 8`] = `"Invalid names list. Array must be non-empty."`;
